dts/arm/96b_carbon.dts

```diff
@@ -8,8 +8,9 @@
 #include <st/stm32f401.dtsi>
 
 / {
-	model = "STMicroelectronics STM32F401RE-NUCLEO board";
-	compatible = "st,stm32f401re-nucleo", "st,stm32f401";
+	model = "Seeed Studio Carbon 96board";
+	compatible = "seeed,carbon", "st,stm32f401re-nucleo",
```

Comment:
The `seeed,carbon` part seems to follow the style used in 96b_nitrogen.dts, but shouldn't this be `compatible = "seeed,carbon", "st,stm32f401"` (that is, without the `st,stm32f401re-nucleo`)? 96b_carbon is not pin-compatible with the Nucleo board.

@erwango, is there a policy on how STM-based boards are named here?